geospaas_processing/utils.py

```diff
@@ -330,3 +332,37 @@ def yaml_env_safe_load(stream):
     """
     yaml.SafeLoader.add_constructor('!ENV', lambda loader, node: os.getenv(node.value))
     return yaml.safe_load(stream)
+
+
+class TrustDomainSession(requests.Session):
+    """Session class which allows keeping authentication headers in
+    case of redirection to the same domain
+    """
+
+    def should_strip_auth(self, old_url, new_url):
+        """Keep the authentication header when redirecting to a
+        different host in the same domain, for example from
+        "scihub.copernicus.eu" to "apihub.copernicus.eu".
+        If not in this case, defer to the parent class.
+        """
+        old_split_hostname = urlparse(old_url).hostname.split('.')
+        new_split_hostname = urlparse(new_url).hostname.split('.')
+        if (len(old_split_hostname) == len(new_split_hostname) > 2
+                and old_split_hostname[1:] == new_split_hostname[1:]):
+            return False
+        else:
+            return super().should_strip_auth(old_url, new_url)
+
+
+def http_request(http_method, *args, **kwargs):
+    """Wrapper around requests.request() which runs the HTTP request
+    inside a TrustDomainSession if authentication is provided. This
+    makes it possible to follow redirections inside the same domain.
+    """
+    auth = kwargs.pop('auth', None)
+    if auth:
+        with TrustDomainSession() as session:
+            session.auth = auth
+            return session.request(http_method, *args, **kwargs)
+    else:
+        return requests.request(http_method, *args, **kwargs)
```

Comment:
We have duplication of code here. Can you place the code in the utils of geospaas so that both harvesting and processing use only one single code?

Reply:
`geospaas_harvesting` and `geospaas_processing` are different packages, so the only way to do that would be to put the utilities in a separate package, which both harvesting and processing would have as a dependency. This means a new repository and CI, version management, etc.
I don't think it's worth it while the common code is that small.

Reply:
No, I mean put the code in the geospaas repo, not another repo. Obviously another repo is not required here for this tiny code.

Reply:
Do you mean putting it in https://github.com/nansencenter/django-geo-spaas/blob/master/geospaas/utils/utils.py ?

Reply:
yes

Reply:
Hum, I don't know what the best solution would be.
Any thoughts on this @akorosov?

Reply:
Let's make a new issue in geospaas for that.